### DISCUSSION TITLE
Set Tuya/MOES TRV comfort_temperature [solved]

### DISCUSSION THREAD

**Discussion by Keeze** [Koenkk/zigbee2mqtt#5548] [Devices]
Hi,

I try to set the comfort preset of my Moes/Tuya TRV to 19C with:

![2021-01-04 (4)](https://user-images.githubusercontent.com/6306958/103551939-65400280-4eab-11eb-95b4-ededf687da9d.png)

It looks like the thermostat does not recognize this command. Am I doing something wrong?

![2021-01-04 (6)](https://user-images.githubusercontent.com/6306958/103552103-acc68e80-4eab-11eb-85b7-7faab0186cdf.png)

I use MQTT Explorer to send the message.

Edit: I also tried from Home Assitant with the MQTT-publish service, but no luck
I was able to set the away_preset_temperature from Home Assistant though

Tnx, Kees

**Comment by Morphy99** [comment]
I had the same trying to set the Eco temperature

**Reply by Keeze** [reply]
Eco doesn't work for me either

**Comment by Morphy99** [comment]
I think your json is incorrect.
Try:
```
  {
      "eco_temperature": 17
  }
```

I figured out mine wasn't working as it wasn't paired properly. I couldn't select any heating mode apart from auto was a giveaway! I thought it was just a crap TRV :D
Now I've removed it and re-paired, it's accepting the set temp.

**Reply by Keeze** [reply]
Also tried that

**Reply by Morphy99** [reply]
Your json in MQTT explorer is as above?
You've tried removing the TRV from Z2M and then resetting and re-pairing?

**Reply by Keeze** [reply]
Yes, JSON is ok, it works when I try to set the away_preset_temperature.

I did not remove and repair the system yet. I will try that tonight, but I doupt that that is gonna help. But I am going to try.

**Reply by Morphy99** [reply]
Well like I said, mine did not accept the commands till I removed and re-paired it. For some reason it wasn't paired correctly as I was missing hvac_modes and the battery always showed empty.

**Reply by Keeze** [reply]
Ok, removed twice and repaired. Now it works, tnx!
